core/src/main/java/io/grpc/transport/Http2ClientStream.java

```diff
@@ -136,8 +136,7 @@ protected void transportDataReceived(Buffer frame, boolean endOfStream) {
       if (endOfStream) {
         // This is a protocol violation as we expect to receive trailers.
         transportError = Status.INTERNAL.withDescription("Recevied EOS on DATA frame");
-        frame.close();
-        inboundTransportError(transportError);
+        inboundTransportError(transportError, true);
```

Comment:
Not stopping delivery was causing the unit test to never report status.  In [transportReportStatus](https://github.com/grpc/grpc-java/blob/2a425090d1fb995464e1df63e88dbaed03635ce1/core/src/main/java/io/grpc/transport/AbstractClientStream.java#L230) when we reach EOS deliveryStalled is false, so we never deliver the status.